tests/file_writer/test_recording.py

```diff
@@ -1724,3 +1660,96 @@ def test_FileWriterProcess__deletes_recorded_stim_data_after_stop_time(
         )
 
         this_file.close()
+
+
+def test_FileWriterProcess__upload_thread_gets_added_to_container_after_all_files_get_finalized(
+    four_board_file_writer_process, mocker
+):
+    update_customer_settings_command = copy.deepcopy(GENERIC_UPDATE_CUSTOMER_SETTINGS)
+    update_customer_settings_command["config_settings"]["auto_delete_local_files"] = False
+    update_customer_settings_command["config_settings"]["auto_upload_on_completion"] = True
+    file_writer_process_ready_for_upload = file_writer_process_with_closed_h5_files_for_upload(
+        four_board_file_writer_process=four_board_file_writer_process,
+        update_customer_settings_command=update_customer_settings_command,
+    )
+    file_writer_process = file_writer_process_ready_for_upload["fw_process"]
+    to_main_queue = file_writer_process_ready_for_upload["to_main_queue"]
+    mocker.patch.object(file_uploader.ErrorCatchingThread, "errors", autospe=True, return_value=True)
```

Comment:
typo `autospec=True`
Should also mock the `start` function and anything else that will help to avoid sporadic failures